### DISCUSSION TITLE
List of good animatable variable fonts (suggestions welcome!)

### DISCUSSION THREAD

**Discussion by stenson** [coldtype/coldtype#82] [General]
`(edited 2023-06-01 for newer timing apis)`

One question that keeps coming up from new Coldtype users is: what are some good variable fonts to animate with? Coldtype does come with two variable fonts built-in (`Font.ColdtypeObviously()` and `Font.MutatorSans()`), but both of those have limitations (only the letters `CDELOPTY` in the former case, and only upper-case ASCII in the latter), so I thought it'd be fun to put together a list of commercial and open-source faces that work well for animation.

In general, a really "animatable" variable font usually has an axis other than `wght` (aka _weight_), which seems to be the most common axis in variable font world. Nothing against `wght`, it’s just not the most visible when it comes to animation. Same goes for `opsz` (optical size). Both can definitely be animated and look great, but just don't _pop_ the way a `wdth` does. Oh and also `slnt` — another great dimension, though without a more visible companion axis, it can be a little harder to leverage in a text animation.

Also in general, the best site for finding variable fonts is [v-fonts.com](https://v-fonts.com/)

Also also in general, a lot of very capable variable fonts are incredibly expensive, which makes sense given how much work they require, but I want to mention that here because there are a lot of commercial variable fonts from established foundries that I've never tried and wouldn't feel great recommending in this thread, given how expensive they are and that I haven't tried them.

One last in general, if you don’t have [FontGoggles](https://fontgoggles.org/) and you’re on a Mac, you should get it! It’s a great way to open variable fonts and look at their axes, _and_ it doesn’t require you install fonts on your computer, (which is something I basically never do because Coldtype just uses the filesystem). Just drag the .ttf onto the window, couldn’t be easier.

I'll start the list with ones that I've personally used & own (listed in alphabetical order):

- __[Ancho](http://beatrizl.com/project/ancho)__
- __[Blimey](https://www.futurefonts.xyz/loveletters/blimey)__
- __[Cheee](https://ohnotype.co/fonts/cheee)__
- __[Clarette](https://www.futurefonts.xyz/into-the-type/clarette)__
- __[CoFo Peshka](https://www.futurefonts.xyz/contrast-foundry/cofo-peshka)__
- __[Digestive](https://ohnotype.co/fonts/digestive)__
- __[Fit](https://djr.com/fit)__
- __[Job Clarendon](https://djr.com/job-clarendon)__
- __[Kinckq](https://thepytefoundry.net/typefaces/kinckq/)__
- __[Konsole](https://www.futurefonts.xyz/elena-schneider/konsole)__
- __[LTR Principia](https://letterror.com/principia/)__
- __[Libido](https://www.herzbergdesign.com/libido)__
- __[MD Nichrome](https://mass-driver.com/typefaces/md-nichrome)__
- __[Mikrobe](https://store.typenetwork.com/foundry/elefonts/fonts/mikrobe)__
- __[Obviously](https://ohnotype.co/fonts/obviously)__
- __[Ohno Fatface](https://ohnotype.co/fonts/fatface)__
- __[Pennypacker](https://cjtype.com/typefaces/pennypacker/)__
- __[Puffling](https://www.futurefonts.xyz/rohernandez/puffling)__
- __[Tweak Display](https://www.futurefonts.xyz/katja-schimmel/tweak-display)__
- __[Vinila](https://plau.co/vinila/)__
- __[WT Zaft 2 A](https://wisetype.nl/collections/wt-zaft)__
- __[Whirly Birdie](https://whirlybirdie.com/)__

And as a little preview, here's a video of all those fonts (though not an exhaustive video of what each of the fonts can do!):

https://github.com/coldtype/coldtype/assets/89770/1f4fe736-b2f0-43c7-ae9f-5b2f5b171c26

(the code to make that video:)

```python
from coldtype import *
from coldtype.interpolation import interp_dict

fonts = [
    ["Obviously", "ObviouslyVar", "https://ohnotype.co/fonts/obviously"],
    ["Tweak Display", "TweakDisplay", "https://www.futurefonts.xyz/katja-schimmel/tweak-display"],
    ["Cheee", "CheeeVar", "https://ohnotype.co/fonts/cheee"],
    ["Puffling", "Puffling.*Var", "https://www.futurefonts.xyz/rohernandez/puffling"], 
    ["Vinila", "Vinila.*Var", "https://plau.co/vinila/"], 
    ["Clarette", "ClaretteGX", "https://www.futurefonts.xyz/into-the-type/clarette"], 
    ["CoFo Peshka", "CoFoPeshkaVar", "https://www.futurefonts.xyz/contrast-foundry/cofo-peshka"], 
    ["Konsole", "Konsolev", "https://www.futurefonts.xyz/elena-schneider/konsole"], 
    ["Job Clarendon", "JobClarendon", "https://djr.com/job-clarendon"],
    ["Ancho", "Ancho", "http://beatrizl.com/project/ancho"],
    ["MD Nichrome", "MDNichrome-Variable", "https://mass-driver.com/typefaces/md-nichrome"],
    ["Ohno Fatface", "OhnoFatfaceVar", "https://ohnotype.co/fonts/fatface"], 
    ["Blimey", "Blimey", "https://www.futurefonts.xyz/loveletters/blimey"], 
    ["Whirly Birdie", "WhirlyBirdie", "https://whirlybirdie.com/"], 
    ["Pennypacker", "Pennypacker", "https://cjtype.com/typefaces/pennypacker/"],
    ["Fit", "^Fit", "https://djr.com/fit"],
    ["Mikrobe", "Mikrobe", "https://store.typenetwork.com/foundry/elefonts/fonts/mikrobe"],
    ["WT Zaft 2 A", "WTZaft2VariableA.ttf", "https://wisetype.nl/collections/wt-zaft"],
    ["Kinckq", "Kinckq-Variable.ttf", "https://thepytefoundry.net/typefaces/kinckq/"],
    ["LTR Principia", "LTRPrincipia-Italic", "https://letterror.com/principia/"],
    ["Libido", "Libido Smooth Var", "https://www.herzbergdesign.com/libido"],
    ["Digestive", "DigestiveVar", "https://ohnotype.co/fonts/digestive"]]

fonts = sorted(fonts, key=lambda x: x[0])

for idx, f in enumerate(fonts):
    display, finder, url = f
    fnt = Font.Find(finder)
    axes = fnt.variations()
    f.extend([
        {k:0 for k in axes.keys()},
        {k:1 for k in axes.keys()}])

    if False:
        print(f"- __[{display}]({url})__") # for markdown list

frames = 24 # <- "speed"

@animation((1920, 1080)
    , timeline=Timeline(frames*len(fonts), 30)
    , bg=0
    , render_bg=1)
def variables(f):
    idx = f.i//frames
    display, finder, _, a, b = fonts[idx]
    e = Timeable(idx*frames, (idx*frames)+frames).at(f.i)
    sample = display[:3]

    return P(
        StSt(sample, finder, 350
            , **interp_dict(e.e("eeo", 1), a, b))
            .align(f.a.r.drop(150, "S"))
            .f(hsl(0.9, 1, 0.7))
            ,
        StSt(sample, finder, fs:=172, **interp_dict(e.e("l", 0), a, b))
            .align(f.a.r.take(w:=230, "W"))
            .f(hsl(0.3, 1, 0.7))
            .rotate(90),
        StSt(sample, finder, fs
            , **interp_dict(e.e("l", 0), b, a))
            .align(f.a.r.take(w, "E"))
            .f(hsl(0.7, 1, 0.7))
            .rotate(-90),
        StSt(f"“{display}”", finder, 120
            , **interp_dict(e.e("ceo", 1), a, b))
            .align(f.a.r.take(200, "S"))
            .f(hsl(0.6, 1, 0.7))
            .append(lambda p: P()
                .line(p.ambit().inset(-50).en)
                .fssw(-1, hsl(0.17, 1, 0.6), 4)))
```

**Comment by arrowtype** [comment]
I put a good deal of thought into making Recursive and Name Sans so that they might be useful in animations!

In Name Sans, the Hairline is very thin, with the goal of allow text to be "revealed" from the center lines.

In Recursive, all glyph forms stay constant throughout styles (e.g. the dollar sign bar never changes), and multiplexed widths allow transitions without changing line length.

**Reply by stenson** [reply]
Awesome! Thanks for adding those to the list!
